### PR TITLE
[java] Add property to ignore interfaces in FieldDeclarationsShouldBeAtStartOfClassRule

### LINKED ISSUE #345: [java] FieldDeclarationsShouldBeAtStartOfClass: Add ability to ignore interfaces
**Rule Set:** design.html#FieldDeclarationsShouldBeAtStartOfClass

**Code Sample demonstrating the issue:**

```java
public interface StartCaptureListener {
    void onStartCapture(boolean shouldUploadMediaToTba);
}

private static final int TAKE_PHOTO_RC = 334; // This should be fine, interfaces can be above fields
```

**Running PMD through:** Gradle

### PR DESCRIPTION
**Pull request by SUPERCILEX**
fixes #345

@jsotuyod My code looks like this right now which is 😢:
```java
public interface StartCaptureListener {
    void onStartCapture(boolean shouldUploadMediaToTba);
}

private static final int TAKE_PHOTO_RC = 334; // NOPMD https://github.com/pmd/pmd/issues/345
private static final List<String> PERMS; // NOPMD https://github.com/pmd/pmd/issues/345
private static final String MEDIA_CREATOR_KEY = "media_creator"; // NOPMD https://github.com/pmd/pmd/issues/345

private WeakReference<Fragment> mFragment; // NOPMD https://github.com/pmd/pmd/issues/345
private TeamHelper mTeamHelper; // NOPMD https://github.com/pmd/pmd/issues/345
private PermissionRequestHandler mPermHandler; // NOPMD https://github.com/pmd/pmd/issues/345
private WeakReference<OnSuccessListener<TeamHelper>> mListener; // NOPMD https://github.com/pmd/pmd/issues/345
private String mPhotoPath; // NOPMD https://github.com/pmd/pmd/issues/345
private boolean mShouldUploadMediaToTba; // NOPMD https://github.com/pmd/pmd/issues/345
```

I'm trying to add that property you mentioned, but I'm not sure how to test my code or build pmd for that matter. (I don't even know if what I wrote compiles 😄)

### CODE REVIEW

**Comment by SUPERCILEX** [comment]
The travis build passed! 😄

**Comment by adangel** [comment]
Awesome! Thanks for contributing!

As for building, all you'll need is maven, and follow this guide: https://github.com/pmd/pmd/blob/master/BUILDING.md

As for unit testing, I'll add a test for this change and reference them here. Then you can see, how this works ;)

File: pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/design/FieldDeclarationsShouldBeAtStartOfClassRule.java [hunk 1]
```diff
@@ -40,6 +40,8 @@
             "Ignore Enum Declarations that precede fields.", true, 1.0f);
     private BooleanProperty ignoreAnonymousClassDeclarations = new BooleanProperty("ignoreAnonymousClassDeclarations",
             "Ignore Field Declarations, that are initialized with anonymous class declarations", true, 2.0f);
+    private BooleanProperty ignoreInterfaceDeclarations = new BooleanProperty("ignoreInterfaceDeclarations",
```
Comment (adangel) [inline pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/design/FieldDeclarationsShouldBeAtStartOfClassRule.java]: While writing the unit test, I figured out, that one piece is missing. The properties need to be declared for the rule. This usually happens in the constructor like this:

    definePropertyDescriptor(ignoreInterfaceDeclarations);

I'll add this. It's always good to have unit tests :)

**Comment by adangel** [comment]
This will be part of PMD 5.7.0

And here's the unit test: https://github.com/adangel/pmd/commit/44bb2826f1590d86814c2a1f9888e0ae083ebf90 
It's basically just additional test cases described in XML. The test cases are run via the [DesignRulesTest](https://github.com/pmd/pmd/blob/master/pmd-java/src/test/java/net/sourceforge/pmd/lang/java/rule/design/DesignRulesTest.java) JUnit test.

**Comment by adangel** [comment]
One more thing: Since this now merged to master and travis has built it (or is still building), it should be included in the [latest snapshot binaries](https://sourceforge.net/projects/pmd/files/pmd/5.7.0-SNAPSHOT/)

You can set this property in your ruleset.xml like this:

~~~
<rule ref="rulesets/java/design.xml/FieldDeclarationsShouldBeAtStartOfClass">
    <properties>
        <property name="ignoreInterfaceDeclarations" value="true"/>
    </properties>
  </rule>
~~~

**Comment by SUPERCILEX** [comment]
@adangel Wow, thanks for all this great information!!! 😄 I'm glad you caught that mistake with not registering the property, and yeah, unit tests are good. 😆 I've never seen them done in XML before, so that's interesting. It looks really simple actually, you just put in some code in the cdata and tell it the number of violations you expect, is that correct?

Thanks again for all your help, being able to build pmd will certainly make it easier to contribute! 😸

**Pull request merged**